utils.py

```diff
@@ -453,3 +454,17 @@ def get_all_language_codes_and_names():
 def unescape_encoded_uri_component(escaped_string):
     """Unescape a string that is encoded with encodeURIComponent."""
     return urllib.unquote(escaped_string).decode('utf-8')
+
+
+cache_slug = None  # pylint: disable=invalid-name
+def get_cache_slug():
+    if cache_slug is None:
+        yaml_file_content = {}
```

Comment:
This is kind of unclear. Why not:

    if feconf.DEV_MODE:
        cache_slug = feconf.CACHE_SLUG_DEV
    else:
        // open yaml file
        // cache_slug = thing read from yaml file

The problem with the way it's done here is that you're shadowing errors. If the yaml file has a typo, the production server will read feconf.CACHE_SLUG as the default. Instead it should fail noisily.